From f1f7e79e50a3e3ba2d891b85a8c75830910523dd Mon Sep 17 00:00:00 2001
From: Marian Buschsieweke <marian.buschsieweke@posteo.net>
Date: Sun, 12 Oct 2025 08:59:10 +0200
Subject: [PATCH] tests/conftest.py: Make pdbp dependency optional

In CI contexts, running the tests without pdbp can be helpful. Therefore,
the dependency to pdbp is made optional.

https://gitlab.com/uweschmitt/pytest-regtest/-/merge_requests/18
---
 tests/conftest.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tests/conftest.py b/tests/conftest.py
index 78f3c9f..12aa764 100644
--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,9 +1,13 @@
 import sys
 
 import numpy
-import pdbp  # noqa
 import pytest
 
+try:
+    import pdbp  # noqa
+except ModuleNotFoundError:
+    pass # pdbp is purely optional here
+
 # hack to disable reload numpy in pytester (current code only prevent reload of
 # zope, so we trick pytester:
 # (see also _pytest/pytester.py  __take_sys_modules_snapshot method)
-- 
2.51.0

